src/main/scala/akka/persistence/jdbc/query/dao/ByteArrayReadJournalDao.scala

```diff
@@ -82,8 +82,15 @@ trait OracleReadJournalDao extends ReadJournalDao {
 
   abstract override def allPersistenceIdsSource(max: Long): Source[String, NotUsed] = {
     if (isOracleDriver(profile)) {
-      Source.fromPublisher(
-        db.stream(sql"""SELECT DISTINCT "#$persistenceId" FROM #$theTableName WHERE rownum <= $max""".as[String]))
+      val selectStatement =
+        if (readJournalConfig.includeDeleted)
+          sql"""SELECT DISTINCT "#$persistenceId" FROM #$theTableName WHERE rownum <= $max""".as[String]
+        else
+          sql"""SELECT DISTINCT "#$persistenceId"
+                 FROM #$theTableName
+                 WHERE rownum <= $max AND deleted = false""".as[String]
```

Comment:
The deleted column can be renamed in the config, so you need to apply the same trick here as with the persistenceId